### PR TITLE
fix octave lines across systems

### PR DESCRIPTION
**Pull request by rettinghaus**
This fixes a small bug ending octave lines on system endings: 

current:
octavebug

proposed:
octavefix

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
   <meiHead>
      <fileDesc>
         <titleStmt>
            <title>Octave shift over system breaks</title>
         </titleStmt>
         <pubStmt>
            <date>2020</date>
         </pubStmt>
      </fileDesc>
   </meiHead>
   <music>
      <body>
         <mdiv>
            <score>
               <scoreDef>
                  <staffGrp>
                     <staffDef n="1" lines="5" clef.shape="G" clef.line="2" />
                  </staffGrp>
               </scoreDef>
               <section>
                  <pb />
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note xml:id="n1" dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                     <octave startid="#n1" endid="#n2" dis="8" dis.place="above" />
                  </measure>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <sb/>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <sb/>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <sb/>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <measure>
                     <staff n="1">
                        <layer n="1">
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                           <beam>
                              <note dur="8" oct.ges="6" oct="5" pname="f" />
                              <note dur="8" oct.ges="6" oct="5" pname="a" />
                              <note dur="8" oct.ges="6" oct="5" pname="g" />
                              <note xml:id="n2" dur="8" oct.ges="6" oct="5" pname="b" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
               </section>
            </score>
         </mdiv>
      </body>
   </music>
</mei>
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +13,7 @@
 * Support for pedal lines (@rettinghaus)
 * Options for controlling output tabs (--output-indent and --output-indent-tab)
 * Option to remove ids in the MEI output (--remove-ids) to be passed to GetMEI with the JS toolkit
+* Fix trmolo tuplets (@rettinghaus)
```
Comment (craigsapp) [inline CHANGELOG.md:16]: tremolo (missing e)

File: src/view_control.cpp [hunk 2]
```diff
@@ -684,12 +684,12 @@ void View::DrawOctave(
...
         TextExtend extend;
         dc->GetSmuflTextExtent(str, &extend);
-        int yCode = (disPlace == STAFFREL_basic_above) ? y1 - extend.m_height : y1;
+        const int yCode = (disPlace == STAFFREL_basic_above) ? y1 - extend.m_height : y1;
         DrawSmuflCode(dc, x1 - extend.m_width, yCode, code, staff->m_drawingStaffSize, false);
         dc->ResetFont();
 
         if (octave->GetLendsym() != LINESTARTENDSYMBOL_none)
-            y2 += (disPlace == STAFFREL_basic_above) ? -extend.m_height : extend.m_height;
+            y2 += (disPlace == STAFFREL_basic_above) ? - extend.m_height : extend.m_height;
```
Comment (craigsapp) [inline src/view_control.cpp:692]: You have not run clang-format?  This is a uniary minus sign which usually has no space after it.

File: src/view_control.cpp [hunk 3]
```diff
@@ -616,7 +616,7 @@ void View::DrawOctave(
     data_OCTAVE_DIS dis = octave->GetDis();
     data_STAFFREL_basic disPlace = octave->GetDisPlace();
 
-    int y1 = octave->GetDrawingY();
+    const int y1 = octave->GetDrawingY();
```
Comment (craigsapp) [inline src/view_control.cpp:619]: Using const is OK, but why?  Const enforces that the variable should not be changed, and there should be an important reason for it not to change which I am not seeing here.  If that is what you want to do, then const is appropriate, and if so, then a comment explaining why it should not change would be useful to add to the code.

**Comment by craigsapp** [comment]
New rendering looks grea!

File: src/view_control.cpp (hunk 2 above)
Comment (rettinghaus) [reply src/view_control.cpp]: correct!

File: CHANGELOG.md (hunk 1 above)
Comment (rettinghaus) [reply CHANGELOG.md]: thanks for spotting this

File: src/view_control.cpp (hunk 3 above)
Comment (rettinghaus) [reply src/view_control.cpp]: correct again. Now it's constant, but we might want to refine it later.

**Pull request merged**